pkg/user/admin/remove_pending.go

```diff
@@ -1,33 +1,165 @@
+	ctx := cliCtx.Context
+	logger := common.GetLogger(cliCtx)
+
+	config, err := readAndValidateRemovePendingAdminConfig(cliCtx, logger)
+	if err != nil {
+		return eigenSdkUtils.WrapError("failed to read and validate user can call config", err)
+	}
+	cliCtx.App.Metadata["network"] = config.ChainID.String()
+	elWriter, err := generator(logger, config)
+	if err != nil {
+		return err
+	}
+
+	receipt, err := elWriter.RemovePendingAdmin(
+		ctx,
+		elcontracts.RemovePendingAdminRequest{
+			AccountAddress: config.AccountAddress,
+			AdminAddress:   config.AdminAddress,
+			WaitForReceipt: true,
+		},
+	)
+	if err != nil {
+		return err
+	}
+	common.PrintTransactionInfo(receipt.TxHash.String(), config.ChainID)
+	return nil
+}
+
+func readAndValidateRemovePendingAdminConfig(
+	cliContext *cli.Context,
+	logger logging.Logger,
+) (*removePendingAdminConfig, error) {
+	accountAddress := gethcommon.HexToAddress(cliContext.String(AccountAddressFlag.Name))
+	adminAddress := gethcommon.HexToAddress(cliContext.String(AdminAddressFlag.Name))
+	ethRpcUrl := cliContext.String(flags.ETHRpcUrlFlag.Name)
+	network := cliContext.String(flags.NetworkFlag.Name)
+	environment := cliContext.String(flags.EnvironmentFlag.Name)
+	if environment == "" {
+		environment = common.GetEnvFromNetwork(network)
+	}
+	signerConfig, err := common.GetSignerConfig(cliContext, logger)
+	if err != nil {
+		// We don't want to throw error since people can still use it to generate the claim
+		// without broadcasting it
+		logger.Debugf("Failed to get signer config: %s", err)
+	}
+
+	chainID := utils.NetworkNameToChainId(network)
+	permissionManagerAddress := cliContext.String(PermissionControllerAddressFlag.Name)
+
+	if common.IsEmptyString(permissionManagerAddress) {
+		permissionManagerAddress, err = common.GetPermissionManagerAddress(utils.NetworkNameToChainId(network))
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	logger.Debugf(
+		"Env: %s, network: %s, chain ID: %s, PermissionManager address: %s",
+		environment,
+		network,
+		chainID,
+		permissionManagerAddress,
+	)
+
+	return &removePendingAdminConfig{
+		Network:                  network,
+		RPCUrl:                   ethRpcUrl,
+		AccountAddress:           accountAddress,
+		AdminAddress:             adminAddress,
+		PermissionManagerAddress: gethcommon.HexToAddress(permissionManagerAddress),
+		SignerConfig:             *signerConfig,
+		ChainID:                  chainID,
+		Environment:              environment,
+	}, nil
+}
+
+func generateRemovePendingAdminWriter(
+	prompter utils.Prompter,
+) func(logger logging.Logger, config *removePendingAdminConfig) (RemovePendingAdminWriter, error) {
+	return func(logger logging.Logger, config *removePendingAdminConfig) (RemovePendingAdminWriter, error) {
+		ethClient, err := ethclient.Dial(config.RPCUrl)
+		if err != nil {
+			return nil, eigenSdkUtils.WrapError("failed to create new eth client", err)
+		}
+		elWriter, err := common.GetELWriter(
+			config.AccountAddress,
```

Comment:
you can just return common.GetELWriter (..) instead of creating vars which you return in next line.